docs/user/implement_env.md

```diff
@@ -0,0 +1,312 @@
+# Adding a New Environment
+
+Garage uses the :class:`Environment` API for all RL environments.
+
+The public API methods of :class:`Environment` are:
+
+| Functions                      | Description                          |
```

Comment:
I think maybe rst table is needed here, as the markdown table doesn't show well in the [generated html](https://garage--1875.org.readthedocs.build/en/1875/user/implement_env.html).